File: SWEA/D2/2001. 파리 퇴치/파리 퇴치.py
T = int(input())

for tc in range(1, T+1):
    N, M = map(int, input().split())  # M x M 크기의 파리채

    arr = [list(map(int, input().split())) for _ in range(N)]

    max_sum = 0

    for i in range(N-M+1):
        for j in range(N-M+1):
            catch_fly = 0
            for k in range(i, i+M):
                for l in range(j, j+M):
                    catch_fly += arr[k][l]

            if catch_fly > max_sum:
                max_sum = catch_fly

    print(f'#{tc} {max_sum}')

File: SWEA/D2/2001. 파리 퇴치/파리 퇴치.java
import java.util.Scanner;

public class Solution {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);

        int T = sc.nextInt();

        for(int tc=1; tc<=T; tc++) {
            int N = sc.nextInt();
            int M = sc.nextInt();

            int[][] arr = new int[N][N];

            for(int i=0; i<N; i++){
                for(int j=0; j<N; j++) {
                    arr[i][j] = sc.nextInt();
                }
            }

            int max_sum = 0;
            for(int i=0; i<=N-M; i++) {
                for(int j=0; j<=N-M; j++) {

                    int sum = 0;
                    // M * M 크기 만큼 탐색
                    for(int k=i; k<i+M; k++) {
                        for(int l=j; l<j+M; l++) {
                            sum += arr[k][l];
                        }
                    }
                    if(sum > max_sum) {
                        max_sum = sum;
                    }
                }
            }
            System.out.println("#" + tc + " " + max_sum);
        }
    }
}
